tests/upload_test.go

```diff
@@ -165,3 +166,129 @@ func uploadImage(portForwardURL, token string, expectedStatus int) error {
+		pv, err = f.CreatePVFromDefinition(utils.NewBlockPVDefinition("local-volume", "1G", nil, "manual", nodeName))
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Verify that PV's phase is Available")
+		err = f.WaitTimeoutForPVReady(pv.Name, 60*time.Second)
+		Expect(err).ToNot(HaveOccurred())
+
+		if pvc != nil {
+			Eventually(func() bool {
+				// Make sure the pvc doesn't still exist. The after each should have called delete.
+				_, err := f.FindPVC(pvc.Name)
+				return err != nil
+			}, timeout, pollingInterval).Should(BeTrue())
+		}
+
+		By("Creating PVC with upload target annotation")
+		pvc, err = f.CreatePVCFromDefinition(utils.UploadBlockPVCDefinition())
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Set up port forwarding")
+		uploadProxyURL, portForwardCmd, err = startUploadProxyPortForward(f)
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		By("Stop port forwarding")
+		if portForwardCmd != nil {
+			err = portForwardCmd.Process.Kill()
+			Expect(err).ToNot(HaveOccurred())
+			portForwardCmd.Wait()
+			portForwardCmd = nil
+		}
+
+		By("Delete upload PVC")
+		err = f.DeletePVC(pvc)
+		Expect(err).ToNot(HaveOccurred())
+		By("Wait for upload pod to be deleted")
+		deleted, err := utils.WaitPodDeleted(f.K8sClient, utils.UploadPodName(pvc), f.Namespace.Name, time.Second*20)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(deleted).To(BeTrue())
+
+		if pv != nil {
+			By("Delete PV for block PV")
+			err := utils.DeletePV(f.K8sClient, pv)
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		if storageClass != nil {
+			By("Delete storageClass for block PV")
+			err = utils.DeleteStorageClass(f.K8sClient, storageClass)
+			Expect(err).ToNot(HaveOccurred())
+		}
+	})
+
+	table.DescribeTable("should", func(validToken bool, expectedStatus int) {
+		By("Verify that upload server POD running")
+		err := f.WaitTimeoutForPodReady(utils.UploadPodName(pvc), time.Second*20)
+		Expect(err).ToNot(HaveOccurred())
+
+		By("Verify PVC status annotation says running")
+		found, err := utils.WaitPVCPodStatusRunning(f.K8sClient, pvc)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(found).To(BeTrue())
+
+		var token string
+		if validToken {
+			By("Get an upload token")
+			token, err = utils.RequestUploadToken(f.CdiClient, pvc)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(token).ToNot(BeEmpty())
+		} else {
+			token = "abc"
+		}
+
+		By("Do upload")
+		err = uploadImage(uploadProxyURL, token, expectedStatus)
+		Expect(err).ToNot(HaveOccurred())
+
+		if validToken {
+			By("Verify PVC status annotation says succeeded")
+			found, err := utils.WaitPVCPodStatusSucceeded(f.K8sClient, pvc)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(found).To(BeTrue())
+
+			same, err := f.VerifyTargetPVCContentMD5(f.Namespace, pvc, utils.DefaultPvcMountPath, utils.UploadBlockDeviceMD5)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(same).To(BeTrue())
+			fileSize, err := f.RunCommandAndCaptureOutput(pvc, "lsblk -n -b -o SIZE \"%s\" /pvc")
```

Comment:
It doesn't like this command, the %s looks suspicious to me, since that a golang string substitution, but that is not what is happening here.

Reply:
changed it